pkg/rootless/rootless_linux.go

```diff
@@ -235,6 +237,24 @@ func becomeRootInUserNS(pausePid, fileToRead string, fileOutput *os.File) (_ boo
 		return false, 0, nil
 	}
 
+	if mounts, err := pmount.GetMounts(); err == nil {
+		for _, m := range mounts {
+			if m.Mountpoint != "/" {
+				continue
+			}
+			isShared := false
+			for _, o := range strings.Split(m.Optional, ",") {
+				if strings.HasPrefix(o, "shared:") {
+					isShared = true
+					break
+				}
+			}
+			if !isShared {
+				logrus.Warningf("%q is not a shared mount, this could cause issues or missing mounts with rootless containers", m.Mountpoint)
+			}
```

Comment:
Should there be a break here? There is no point in looping over the remaining mount points.

Reply:
thanks for spotting it.  Fixed and pushed a new version